http/client.go

```diff
@@ -427,6 +427,44 @@ func (c *InternalClient) marshalImportPayload(index, field string, shard uint64,
 	return buf, nil
 }
 
+// Forward the request to the new host in request
+func (c *InternalClient) Forward(ctx context.Context, w http.ResponseWriter, req *http.Request, scheme, host string) error {
+
+	body, err := ioutil.ReadAll(req.Body) // it appears I must consume the
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return nil
+	}
+
+	req.Body = ioutil.NopCloser(bytes.NewReader(body))
+	url := fmt.Sprintf("%s://%s%s", scheme, host, req.RequestURI)
+
+	forReq, err := http.NewRequest(req.Method, url, bytes.NewReader(body))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return nil
+	}
+
+	forReq.Header = make(http.Header)
+	for h, val := range req.Header {
+		forReq.Header[h] = val
+	}
```

Comment:
I think it's possible to forward the original request without creating a copy (maybe without reading from body?)

In the PDK proxy I used the same object, but had to do a few things:
```
	orig.URL = reqURL // set to new URL
	orig.Host = ""    // can/should be empty for client requests
	orig.RequestURI = "" // should not be set in a client request
	orig.ContentLength = int64(len(origbody)) // can probably leave this alone if body is untouched.
```
I had to read the body in the PDK because I was modifying it, but it'd be cool if we could avoid it.